_pytest/__init__.py

```diff
@@ -6,3 +6,5 @@
     # broken installation, we don't even try
     # unknown only works because we do poor mans version compare
     __version__ = 'unknown'
+
+from .vendored_packages import attr
```

Comment:
I'm a little wary of this approach of importing a module simply so it's exposed for other modules to import. It may work for `attrs` because attrs happens to import all of its submodules, but in the general case, this pattern doesn't work. I considered something like this when vendoring packaging into setuptools, but packaging doesn't import all of its submodules, so an `from ._vendor import packaging` doesn't expose `packaging.specifiers` (or simililar), but moreover, it's difficult for another module that needs those submodules to make the submodules available.

This pattern is discouraged for exposing non-vendored functionality, so I think it should be discouraged for vendored as well.